include/nanorange/algorithm/stable_partition.hpp

```diff
@@ -74,13 +77,13 @@ struct stable_partition_fn {
             if (nano::invoke(pred, nano::invoke(proj, *middle))) {
                 nano::iter_swap(first, middle);
                 nano::iter_swap(middle, last);
-                return last;
+                return {first, nano::next(nano::next(first))};
```

Comment:
This `next(next(first))` thing looks weird. I tried `advance(first, 2)`, but that resulted in `first` having a different type than `advance(first)`, which resulted in compilation error, as the return value is `subrange<I>`.